assets/js/prot_struct.js

```diff
@@ -6,6 +6,44 @@ stage.setParameters({
   backgroundColor: "white"
 });
 
+// color scheme
+function createProteinColorScheme(targetChains){
+  targetChains = ":" + targetChains.join(" or :");
```

Comment:
If the user's data somehow produces a list of target chains with an empty string, this line produces an invalid query for NGL (I've included the full description in the main review text).

This line could be changed to something like:

```js
targetChains = ":" + targetChains.filter((item) => item != "").join(" or :");
```

Reply:
I don't believe there should ever be a empty string chain for the protein structure. I filtered out empty strings from the protein chains list when the list is created rather than each time a new color scheme is created.